www/islands/SearchButton.tsx

```diff
@@ -1,12 +1,21 @@
 import { Head } from "$fresh/runtime.ts";
 import { useEffect, useRef } from "preact/hooks";
-import docsearch from "https://esm.sh/@docsearch/js@3?target=es2020";
 
-export default function SearchButton(props: { class?: string }) {
+// Copied from algolia source code
+type DocSearchProps = {
+  appId: string;
+  apiKey: string;
+  indexName: string;
+  container: HTMLElement | string;
+};
+
+export default function SearchButton(
```

Comment:
The `docsearch` prop was added to allow mocking the function from a test (`DocSearchTest`). The `DocsSidebar` component was modified to send in the real `docsearch` implementation using this prop.